Description: Workaround a race condition in shtool's mkdir -p
 This workaround can be affected by another race condition where a
 different process running under a different user creates the directory,
 which would make the chown/chmod calls fail.
 .
 This is the version of the patch sent to shtool upstream.
Origin: vendor
Forwarded: yes
Last-Update: 2010-02-18

--- a/build/shtool
+++ b/build/shtool
@@ -1003,7 +1003,11 @@ mkdir )
                     if [ ".$opt_t" = .yes ]; then
                         echo "mkdir $pathcomp" 1>&2
                     fi
-                    mkdir $pathcomp || errstatus=$?
+                    mkdir $pathcomp || {
+                        _errstatus=$?
+                        [ -d "$pathcomp" ] || errstatus=${_errstatus}
+                        unset _errstatus
+                    }
                     if [ ".$opt_o" != . ]; then
                         if [ ".$opt_t" = .yes ]; then
                             echo "chown $opt_o $pathcomp" 1>&2
